File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/domain/repository/RecipeRepository.java
package com.tarbadev.witchcraft.recipes.domain.repository;

import com.tarbadev.witchcraft.recipes.domain.entity.Recipe;

import java.util.List;

public interface RecipeRepository {
  Recipe saveRecipe(Recipe recipe);

  Recipe updateRecipe(Recipe recipe);

  List<Recipe> findAll();

  Recipe findById(Integer id);

  void delete(int id);

  void setFavorite(int id, Boolean isFavorite);

  List<Recipe> findAllFavorite();

  List<Recipe> findLastAddedRecipes();

  Boolean existsById(Integer id);
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/recipes/domain/repository/RecipeRepository.kt
package com.tarbadev.witchcraft.recipes.domain.repository

import com.tarbadev.witchcraft.recipes.domain.entity.Recipe

interface RecipeRepository {
    fun save(recipe: Recipe): Recipe

    fun update(recipe: Recipe): Recipe

    fun findAll(): List<Recipe>

    fun findById(id: Int): Recipe?

    fun delete(id: Int)

    fun setFavorite(id: Int, isFavorite: Boolean): Recipe

    fun findAllFavorite(): List<Recipe>

    fun findLastAddedRecipes(): List<Recipe>

    fun existsById(id: Int): Boolean
}
